chainerx_cc/examples/mnist_cc/mnist.cc

```diff
@@ -0,0 +1,85 @@
+#include "mnist.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+namespace {
+
+template <typename T>
+T ReadValue(std::ifstream& ifs) {
+    T val{0};
+    ifs.read(reinterpret_cast<char*>(&val), sizeof(val));
+    return val;
+}
+
+chainerx::Array ReadArray(std::ifstream& ifs, const chainerx::Shape& shape) {
+    int64_t n = shape.GetTotalSize();
+
+    std::shared_ptr<uint8_t> data{new uint8_t[n], std::default_delete<uint8_t[]>{}};
+    std::generate_n(data.get(), n, [&ifs]() { return ReadValue<uint8_t>(ifs); });
```

Comment:
You can simply write `ifs.read(data.get(), n)`.

Reply:
Thanks, fixed.